config/config_handler.go

```diff
@@ -35,23 +36,21 @@ type ConfHandler interface {
 	Start()
 	Close()
 	GetConfig() *Config // read only
+	SetConfig(conf *Config) error
 }
 
 // ConfReloadFunc is used to reload the config to make it work.
 type ConfReloadFunc func(oldConf, newConf *Config)
 
 // NewConfHandler creates a new ConfHandler according to the local config.
 func NewConfHandler(localConf *Config, reloadFunc ConfReloadFunc) (ConfHandler, error) {
-	switch defaultConf.Store {
-	case "tikv":
+	if strings.ToLower(localConf.Store) == "tikv" && localConf.EnableDynamicConfig {
 		return newPDConfHandler(localConf, reloadFunc, nil)
-	default:
-		return &constantConfHandler{localConf}, nil
 	}
+	return &constantConfHandler{localConf}, nil
 }
 
 // constantConfHandler is used in local or debug environment.
```

Comment:
I think `used in local` is not explicit, we need to mention in this comment that when `EnableDynamicConfig` is false, this will be used, and the `constantConfHandler.conf` will always be the configuration that initialized when tidb-server is started.